spring-cloud-starter-tencent-polaris-discovery/src/main/java/com/tencent/cloud/polaris/registry/PolarisServiceRegistry.java

```diff
@@ -138,42 +129,37 @@ public void register(PolarisRegistration registration) {
 		try {
 			ProviderAPI providerClient = polarisSDKContextManager.getProviderAPI();
 			InstanceRegisterResponse instanceRegisterResponse;
-			if (StringUtils.isBlank(polarisDiscoveryProperties.getHealthCheckUrl())) {
-				instanceRegisterResponse = providerClient.registerInstance(instanceRegisterRequest);
-			}
-			else {
-				instanceRegisterResponse = providerClient.register(instanceRegisterRequest);
-				InstanceHeartbeatRequest heartbeatRequest = new InstanceHeartbeatRequest();
-				BeanUtils.copyProperties(instanceRegisterRequest, heartbeatRequest);
-				heartbeatRequest.setInstanceID(instanceRegisterResponse.getInstanceId());
-				// Start the heartbeat thread after the registration is successful.
-				heartbeat(heartbeatRequest);
+			
+			// Always use registerInstance to avoid automatic heartbeat
+			instanceRegisterRequest.setTtl(polarisDiscoveryProperties.getHeartbeatEnabled() ? 
```

Comment:
I think you should modify com.tencent.polaris.api.core.ProviderAPI#registerInstance but not set a big heartbeat TTL.